analyzers/src/SonarAnalyzer.Common/Rules/UseUnixEpochBase.cs

```diff
@@ -0,0 +1,116 @@
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules;
+
+public abstract class UseUnixEpochBase<TSyntaxKind, TLiteralExpression, TMemberAccessExpression> : SonarDiagnosticAnalyzer<TSyntaxKind>
+    where TSyntaxKind : struct
+    where TLiteralExpression : SyntaxNode
+    where TMemberAccessExpression : SyntaxNode
+{
+    private const string DiagnosticId = "S6588";
+
+    protected override string MessageFormat => "Use \"{0}.UnixEpoch\" instead of creating {0} instances that point to the unix epoch time";
+
+    protected abstract bool IsDateTimeKindUtc(TMemberAccessExpression memberAccess);
+    protected abstract bool IsGregorianCalendar(SyntaxNode node);
+    protected abstract bool IsZeroTimeOffset(SyntaxNode node);
+
+    protected UseUnixEpochBase() : base(DiagnosticId) { }
+
+    protected sealed override void Initialize(SonarAnalysisContext context) =>
+        context.RegisterCompilationStartAction(start =>
+        {
+            if (!CompilationTargetsValidNetVersion(start.Compilation))
+            {
+                return;
+            }
+
+            context.RegisterNodeAction(
+                Language.GeneratedCodeRecognizer,
+                c =>
+                {
+                    var argumentCount = Language.Syntax.ArgumentExpressions(c.Node).Count();
+                    if (argumentCount < 3)
+                    {
+                        return;
+                    }
+
+                    var type = c.SemanticModel.GetTypeInfo(c.Node).Type;
+                    if (type.DerivesFrom(KnownType.System_DateTime) && IsEpochCtor(c.Node, c.SemanticModel))
+                    {
+                        c.ReportIssue(Diagnostic.Create(Rule, c.Node.GetLocation(), "DateTime"));
+                    }
+                    else if (type.DerivesFrom(KnownType.System_DateTimeOffset) && IsEpochCtor(c.Node, c.SemanticModel))
+                    {
+                        c.ReportIssue(Diagnostic.Create(Rule, c.Node.GetLocation(), "DateTimeOffset"));
+                    }
+                },
+                Language.SyntaxKind.ObjectCreationExpressions);
+        });
+
+    protected static bool IsValueEqualsTo(TLiteralExpression literal, int value) =>
+        literal.ChildTokens().First().ValueText == value.ToString();
```

Comment:
Based on this comment: instead of checking for string equality, use `int.TryParse()` and compare the actual integer value. Then add test cases for 01 and 0x1.